tests/ui/associated-types/dont-suggest-cyclic-constraint.stderr

```diff
@@ -0,0 +1,12 @@
+error[E0308]: mismatched types
+  --> $DIR/dont-suggest-cyclic-constraint.rs:7:35
+   |
+LL |     debug_assert_eq!(iter.next(), Some(value));
+   |                                   ^^^^^^^^^^^ expected `Option<<I as Iterator>::Item>`, found `Option<&<I as Iterator>::Item>`
+   |
+   = note: expected enum `Option<<I as Iterator>::Item>`
+              found enum `Option<&<I as Iterator>::Item>`
```

Comment:
The expectation in the ticket was for this to suggest `.as_ref()`, right?

Reply:
Yes, which is why I didn't close it just yet.